common/ResetPasswordCore.py

```diff
@@ -0,0 +1,177 @@
+        user's ID + random generated 8 digit code."""
+        id = DbHandler.get_user_id(username)
+        if id == -1:
+            return "USER_NOT_FOUND"
+
+        random_code = ResetPasswordCore.__generate_8_digit_code()
+        forget_password_rendered_page = ResetPasswordCore.\
+            __render_forget_password_page(random_code)
+
+        ResetPasswordCore.__send_email_to_user(
+            username,
+            'Restfulness Forget Password Code',
+            forget_password_rendered_page)
+
+        return(ResetPasswordCore.__generate_hash_string(id, random_code))
+
+    @staticmethod
+    def get_password_reset_token(hashed_data: str, user_input: str) -> str:
+        """ Step 2: Return token for reseting password if user inputed correct
+        8 digit code."""
+        try:
+            user_id = ResetPasswordCore.__verify_user_input_with_hashed_data(
+                hashed_data, user_input)
+        except SignatureExpired:
+            return('EXPIRED')
+        except BadSignature:
+            return('INVALID_TOKEN')
+
+        if user_id == -1:
+            return('INVALID_CODE')
+        else:
+            return(ResetPasswordCore.__generate_reset_password_token(user_id))
+
+    @staticmethod
+    def reset_password(reset_token: str, new_password: str) -> str:
+        """ Step 3: Change user's password if the token for changing
+        password is valid"""
+        try:
+            user_id = ResetPasswordCore.__validate_reset_password_token(
+                reset_token)
+        except SignatureExpired:
+            return('EXPIRED')
+        except BadSignature:
+            return('INVALID_TOKEN')
+
+        status = DbHandler.reset_user_forgotten_password(user_id, new_password)
+        if status == 'OK':
+            return('OK')
+        else:
+            return('FAILED')
+
+    @staticmethod
+    def __validate_reset_password_token(token: str) -> int:
+        """ Return user's ID if reset password token is correct. """
+        hash = Serializer(SECRET_KEY)
+        try:
+            data = hash.loads(token)
+        except SignatureExpired:
+            raise
+        except BadSignature:
+            raise
+
+        return(data['id'])
+
+    @staticmethod
+    def __verify_user_input_with_hashed_data(hashed_data: str,
+                                             user_input: str) -> int:
+        """ Verify if user's inputed 8 digit code is correct, if so,
+        return users id."""
+        hash = Serializer(SECRET_KEY)
+        try:
+            data = hash.loads(hashed_data)
+        except SignatureExpired:
+            raise
+        except BadSignature:
+            raise
+
+        if user_input == data['valid_code']:
+            return(data['id'])
+        else:
+            return(-1)
+
+    @staticmethod
+    def __generate_reset_password_token(id: int) -> str:
+        """ Generate the main token that reset user's password."""
+        hash = Serializer(SECRET_KEY, expires_in=300)
```

Comment:
The expiration time should be configurable and also the default value should mentioned in the readme file!